src/solvers/flattening/boolbv_unary_minus.cpp

```diff
@@ -23,66 +25,53 @@ bvt boolbvt::convert_unary_minus(const unary_exprt &expr)
   if(width==0)
     return conversion_failed(expr);
 
-  const exprt::operandst &operands=expr.operands();
-
-  if(operands.size()!=1)
-    throw "unary minus takes one operand";
-
-  const exprt &op0=expr.op0();
+  const exprt &op = expr.op();
 
-  const bvt &op_bv=convert_bv(op0);
+  const bvt &op_bv = convert_bv(op, width);
 
   bvtypet bvtype=get_bvtype(type);
-  bvtypet op_bvtype=get_bvtype(op0.type());
-  std::size_t op_width=op_bv.size();
+  bvtypet op_bvtype = get_bvtype(op.type());
 
   bool no_overflow=(expr.id()=="no-overflow-unary-minus");
```

Comment:
`"no-overflow-unary-minus"` does not seem to be assigned anywhere. Is this still needed?

Reply:
@peterschrammel Do you know if we can remove "no-overflow-unary-minus"?